mitiq/adaptive_zne.py

```diff
@@ -0,0 +1,102 @@
+from pyquil import Program
+from typing import List, Tuple
+
+from mitiq.zne import richardson_extr
+import mitiq.qiskit.qiskit_utils as qs_utils
+
+
+class Generator(object):
+    """
+    This object adaptively generates new noise scaling parameters based on a historical stack of previous noise
+    scale parameters and previously calculated expectation values. Noise scaling parameters are stored on the `in_stack`
+    and the running list of expectation values are stored on the `out_stack`.
+    """
+    def __init__(self, instack: List[float], outstack: List[float]):
```

Comment:
What is our style guide for `typing` and function signatures? I vote to specify return types for all functions even `__init__`, meaning this would be changed to

```python
def __init__(self, instack: List[float], outstack: List[float]) -> None:
```

Reply:
Sounds good to me +1

Reply:
added types